gsy_framework/community_datasheet/community_datasheet_validator.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from collections import Counter
+from datetime import timedelta
+
+from jsonschema import Draft202012Validator
+from jsonschema.exceptions import ValidationError
+from jsonschema.validators import extend
+
+from gsy_framework.community_datasheet.community_datasheet_reader import CommunityDatasheet
+from gsy_framework.community_datasheet.exceptions import CommunityDatasheetException
+from gsy_framework.community_datasheet.schema import COMMUNITY_DATASHEET_SCHEMA
+from gsy_framework.constants_limits import FIELDS_REQUIRED_FOR_REBASE
+from gsy_framework.scenario_validators import scenario_validator
+
+logger = logging.getLogger(__name__)
+
+
+class CommunityDatasheetValidator:
+    """Validator for the parsed community datasheet output."""
+
+    def __init__(self):
+        # JSON schema doesn't work with datetime or custom objects, so we customise it
+        type_checker = Draft202012Validator.TYPE_CHECKER.redefine(
+            "custom_timedelta", self._is_timedelta)
+        CustomValidator = extend(Draft202012Validator, type_checker=type_checker)
+        self._validator = CustomValidator(schema=COMMUNITY_DATASHEET_SCHEMA)
+
+    def validate(self, datasheet: CommunityDatasheet):
+        """Validate the JSON output of the datasheet."""
+        try:
+            self._validator.validate(instance=datasheet.as_dict())
+        except ValidationError as ex:
+            raise CommunityDatasheetException(ex) from ex
+
+        self._validate_loads(datasheet)
+        self._validate_pvs(datasheet)
+        self._validate_unique_names(datasheet)
+        self._validate_grid(datasheet.grid)
+
+    @staticmethod
+    def _is_timedelta(_checker, instance):
+        return isinstance(instance, timedelta)
+
+    @staticmethod
+    def _validate_unique_names(datasheet):
```

Comment:
This is only for mapping the profiles to the correct assets, right? From a simulations' view it only makes sense to omit duplicated names inside one market area.